air/src/air/mod.rs

```diff
@@ -283,20 +296,50 @@ pub trait Air: Send + Sync {
     #[allow(unused_variables)]
     fn get_aux_assertions<E: FieldElement<BaseField = Self::BaseField>>(
         &self,
-        aux_rand_elements: &AuxTraceRandElements<E>,
+        aux_rand_elements: &[E],
     ) -> Vec<Assertion<E>> {
         Vec::new()
     }
 
+    // AUXILIARY PROOF VERIFIER
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the [`GkrVerifier`] to be used to verify the GKR proof.
+    ///
+    /// Leave unimplemented if the `Air` doesn't use a GKR proof.
+    fn get_auxiliary_proof_verifier<E: FieldElement<BaseField = Self::BaseField>>(
+        &self,
+    ) -> Self::GkrVerifier {
+        unimplemented!("`get_auxiliary_proof_verifier()` must be implemented when the proof contains a GKR proof");
+    }
```

Comment:
Looking at this now, I wonder if we need the `GkrVerifier` trait at all. Maybe a single function here - e.g., `verify_gkr_proof()` would have sufficed.

Reply:
Agreed. IMO we should wait for the next PR, as this will change anyways